src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,133 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
+@SuppressWarnings("unchecked")
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] array;
+    private int size;
+
+    public ArrayList() {
+        array = new Object[DEFAULT_CAPACITY];
+    }
+
+    public ArrayList(int capacity) {
+        if (capacity > 0) {
+            array = new Object[capacity];
+        } else {
+            throw new IllegalArgumentException("Capacity must be greater than zero.");
+        }
+    }
+
     @Override
     public void add(T value) {
-
+        if (size >= array.length) {
+            grow();
+        }
+        array[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
```

Comment:
The condition `index > size` should be `index >= size` to prevent adding at an index equal to the current size, which is valid for appending.

Reply:
My unit tests are cheking cases, where 'index == size' and they are succesful, so I should maintain only 'bigger than size cases' it in order to pass unit tests